pycytominer/cyto_utils/cells.py

```diff
@@ -436,27 +437,21 @@ def aggregate_compartment(
         ).rename(self.linking_col_rename, axis="columns")
 
         # Infering features is tricky with non-canonical data
-        if aggregate_args is None:
-            aggregate_args = {}
-            features = infer_cp_features(population_df, compartments=compartment)
-        elif "features" not in aggregate_args:
-            features = infer_cp_features(population_df, compartments=compartment)
-        elif aggregate_args["features"] == "infer":
-            features = infer_cp_features(population_df, compartments=compartment)
+        if self.features == "infer":
```

Comment:
Ah, this is so much nicer... You can remove line 439 - it is no longer tricky!

Reply:
:) Done